org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/internal/DefaultModelPersistence.java

```diff
@@ -54,32 +52,33 @@
 public final class DefaultModelPersistence implements ModelPersistence, EventListener {
 
     // Cache holding (project name - persistent model property) pairs.
-    private final LoadingCache<IProject, PersistentModel> modelCache;
+    private final Map<IProject, PersistentModel> cache;
 
     private DefaultModelPersistence() {
-        this.modelCache = CacheBuilder.newBuilder().build(new CacheLoader<IProject, PersistentModel>() {
-
-            @Override
-            public PersistentModel load(IProject project) throws Exception {
-                return doLoadModel(project);
-            }
-        });
+        this.cache = Maps.newConcurrentMap();
     }
 
     @Override
     public PersistentModel loadModel(IProject project) {
-        return this.modelCache.getUnchecked(project);
+        PersistentModel model = this.cache.get(project);
+        if (model == null) {
+            model = readModel(project);
+            if (model != null) {
+                this.cache.put(project, model);
```

Comment:
I reverted to the `CacheLoader` implementation and used a new `AbsentPersistentModel` class to distinguish nonexisting models.